### DISCUSSION TITLE
DNSCrypt/dnscrypt-proxy can't encrypt tls sni

### DISCUSSION THREAD

**Discussion by lkj0000** [DNSCrypt/dnscrypt-proxy#2213] [Q&A]
DNSCrypt/dnscrypt version: 2.1.2
OS: windows10 21H2
Browser version: 105.0.1343.42
Browser open ECH setting reference: https://techcommunity.microsoft.com/t5/discussions/you-can-now-enable-encrypted-client-hello-encrypted-sni-or-esni/m-p/3600372
Description:
No matter how to set it can not pass the EHC test, you can crawl tls sni through Wireshark to get the real domain access, please ask sni can not encrypt is DNSCrypt/dnscrypt or browser problems?

**Comment by jedisct1** [comment]
ECH happens at the browser level.

You need to be using Cloudflare as a resolver. 

I don't know much about Edge, but maybe it does the same stupid thing as Firefox and requires a direct connection to a DoH server in order to enable ECH. In that case, you'll have to do the same thing as with Firefox: https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Local-DoH

**Reply by jedisct1** [reply]
Also, since Edge is Chromium-based, may you need to [explicitly enable it](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Local-DoH#using-ech-in-chrome).

**Reply by lkj0000** [reply]
Thanks!
The problem I had when checking the wiki was 
The original example." openssl req -x509 -nodes -newkey rsa:2048 -days 5000 -sha256 -keyout \
  localhost.pem -out localhost.pem"

I want to change rsa:2048 to ECC384, then I can't output keypem
Change this example." openssl req -x509 -nodes -newkey ecc:384 -days 365 -sha256 -keyout \
 localhost.pem -out localhost.pem"
 Please guide me, thanks!

**Reply by jedisct1** [reply]
https://www.scottbrady91.com/openssl/creating-elliptical-curve-keys-using-openssl

**Reply by lkj0000** [reply]
Thank you!
The configuration according to https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Local-DoH worked.
The chorme settings are successful but still can't encrypt tls sni , thanks. 
Thank you.
